weave-js/src/components/PagePanel.tsx

```diff
@@ -118,6 +122,12 @@ const usePoorMansLocation = () => {
     const interval = setInterval(() => {
       if (window.location.toString() !== location) {
         setLocation(window.location.toString());
+        const options = {
```

Comment:
can we make a fresh use-hook... something like `usePageAnalytics`. This may call `usePoorMansLocation` internally, or might be even better to use `useHistory`. Either way, it belongs as a reactive dep of location, not in the location hook.

Reply:
sg, will do